ui/store/actions.ts

```diff
@@ -1837,6 +1842,9 @@ export function updateMetamaskState(
       type: actionConstants.UPDATE_METAMASK_STATE,
       value: newState,
     });
+    if (!isEqual(oldSelectedAccountBalance, newSelectedAccountBalance)) {
+      dispatch(initializeSendState({ chainHasChanged: true }));
+    }
     if (provider.chainId !== newProvider.chainId) {
```

Comment:
This is the line where we are already checking if the chainId has changed.

Reply:
Well for some reason the state is not being updated, specifically account balance, so that is why I added when balances are not equal when they should be, update the state and I was advised by @danjm .

Reply:
I am pretty sure what is happening here is a race condition where it takes longer for the balance to update than the network. So at the time of the `dispatch(initializeSendState({ chainHasChanged: true }));` that Brad is referring to, the selected account balance has not been updated.

My recommendation here is that we create a different parameter `balanceHasChanged` and add appropriate handling within the send duck. While this will handle the race condition, it also handles the case where the user's balance changes while they are on the send screen.

So we need the above line to be `dispatch(initializeSendState({ balanceHasChanged: true }));`

Reply:
Okay, no problem I could implement that, but should i proceed with this or should I wait for that discussion that Barbara mentioned in the comments at the end? @danjm @bschorchit